code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -840,31 +840,25 @@
 		if("highfive")
 			if(restrained())
 				return
-			if(EFFECT_HIGHFIVE in active_effect)
-				to_chat(src, "You give up on the high-five.")
-				active_effect -= EFFECT_HIGHFIVE
-				return
-			active_effect |= EFFECT_HIGHFIVE
-			for(var/mob/living/carbon/C in orange(1))
-				if(EFFECT_HIGHFIVE in C.active_effect)
-					if((C.mind.special_role == SPECIAL_ROLE_WIZARD) && (mind.special_role == SPECIAL_ROLE_WIZARD))
+			visible_message("<b>[name]</b> requests a highfive.", "You request a highfive.")
+			if(!has_status_effect(STATUS_EFFECT_HIGHFIVE))
+				apply_status_effect(STATUS_EFFECT_HIGHFIVE)
+
+			if(has_status_effect(STATUS_EFFECT_HIGHFIVE))
```

Comment:
I think this is in the wrong place, shouldn't it be under the for loop? As it is, this will never fail...

Reply:
yeah i had trouble with that...Got in insight on how to restructure it?

Reply:
```dm
for(var/mob/living/carbon/C in orange(1))
    if(!C)
        visible_message("[name was left hanging....
    if(C.has_status_effect(STATUS_EFFECT_HIGHFIVE))
        // Wizard code
```

Reply:
tried it mofied it...didn't trigger the left hanging part...dunno why